Move kubevirtci k8s startup and dev guide to parent dir (#673)
kubevirt/kubevirtci: cluster-up/cluster/K8S_DEV_GUIDE.md

Upon finishing deployment of a K8s deploy, we will have 3 containers:

> * k8s-1.21 vm container - a container that runs a qemu VM, which is the K8s node, in which the pods will run.

Reviewer: ```suggestion
* k8s-${KUBEVIRT_PROVIDER_VERSION} vm container - a container that runs a qemu VM, which is the K8s node, in which the pods will run.
```

```

> # Flow of K8s provisioning (1.21 for example)

Reviewer: ```suggestion
# Flow of K8s provisioning ${KUBEVIRT_PROVIDER_VERSION}
```

> # Flow of K8s cluster-up (1.21 for example)

Reviewer: ```suggestion
# Flow of K8s cluster-up ${KUBEVIRT_PROVIDER_VERSION}
```

kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f "$custom_manifest"
```
* Run ./cluster-provision/k8s/1.21.0/provision.sh, it will create a new provision and test it.

# Manual steps for publishing a new provider

The steps to create, test and integrate a new KubeVirtCI provider are [mostly automated](./K8S_AUTOMATION.md), but just in case you need to do it manually:

> * Run `./cluster-provision/k8s/1.21.0/publish.sh`, it will publish the new created image to quay.io

Reviewer 2: maybe parameterize 1.21.0 ?
(its 1.21 now btw)
